src/pull-progress-indicator.js

```diff
@@ -0,0 +1,103 @@
+const logUpdate = require('log-update');
+const { green, cyan, yellow } = require('colorette');
+const emoji = require('node-emoji');
+
+/**
+ * Some status names are long; this map has shorter names!
+ */
+const SHORT_NAME_STATUS = {
+    'pulling fs layer': 'Preparing',
+    'download complete': 'Downloaded',
+    'pull complete': 'Completed',
+    'retry': 'Retrying'
+};
+
+/**
+ * Maximum column length for status 
+ */
+const STATUS_STRING_MAX_LENGTH = 25;
+
+/**
+ * Cached emoji, so it needn't be fetched repeatedly!
+ */
+const PACKAGE_EMOJI = emoji.get('package');
+
+
+/**
+ * Starts up a progress bar for the `docker pull` operation, parses the event stream from dockerode to get the progress info.
+ */
+class PullProgressIndicator {
+    constructor() {
+        this._chunkInfo = null;
+    }
+
+    /**
+     * Gets shortened status name, because some status names can be a wee bit long.
+     * 
+     * @param {string} status 
+     */
+    _getShortenedStatus(status) {
+        if (!status) return '';
+
+        // Special case for retries
+        if (status.includes('retry')) return SHORT_NAME_STATUS.retry;
+
+        return SHORT_NAME_STATUS[status.toLowerCase()] || status;
+    }
+
+    /**
+     * Uses `log-update` to update the progress information on the terminal
+     */
+    _redrawProgress() {
+        if (this._chunkInfo === null) {
+            // TODO: Should we persist this? https://www.npmjs.com/package/log-update#logupdatedone
+            return logUpdate.clear();
+        }
+
+        const chunkProgress = Object.keys(this._chunkInfo).map(id => {
```

Comment:
My approach consists of leaving the image chunks as-is and show individual status:

```
📦 092586df9206 | Downloading               | [=======================================>           ]  36.29MB/45.38MB
📦 ef599477fae0 | Downloaded                | 
📦 4530c6472b5d | Downloaded                | 
📦 d34d61487075 | Downloading               | [=========================================>         ]  41.21MB/50.07MB
📦 87fc2710b63f | Downloading               | [=====>                                             ]  25.22MB/214.8MB
📦 e83c771c5387 | Waiting                   | 
📦 fef6195892ae | Waiting                   | 
📦 70e1a68db517 | Waiting                   | 
📦 c2599bbef3cb | Waiting                   | 
```

Were you thinking of a more collated progress bar?

Reply:
FYI: Will review this later today. Have to check how it looks like if an image has already been downloaded before. Will follow-up.

Reply:
Just checked locally. Looks good! :)